snekbox/api/resources/eval.py

```diff
@@ -122,10 +124,21 @@ def on_post(self, req: falcon.Request, resp: falcon.Response) -> None:
         if "input" in body:
             body.setdefault("args", ["-c"])
             body["args"].append(body["input"])
+
+        binary_path = body.get("binary_path")
+        if binary_path:
+            binary_path = Path(binary_path)
+            if (
+                not binary_path.resolve().as_posix().startswith("/lang/")
+                or not binary_path.is_file()
```

Comment:
Should we also check that it's executable?
```suggestion
                or not binary_path.is_file()
                or not binary_path.stat().st_mode & 0o100 == 0o100
```

Reply:
Yes! https://github.com/python-discord/snekbox/pull/196/commits/81573b2d0144ece5490948706918233b887f86c1